### PR TITLE
adds rollout path for chained Merkle shreds

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Roll out chained Merkle shreds

#### Summary of Changes
The commit adds `should_chain_merkle_shreds` to incrementally roll out chained Merkle shreds to clusters.

### CODE REVIEW

File: turbine/src/broadcast_stage/standard_broadcast_run.rs [hunk 1]
```diff
@@ -497,10 +527,15 @@ impl BroadcastRun for StandardBroadcastRun {
     }
 }
 
+fn should_chain_merkle_shreds(_slot: Slot, _cluster_type: ClusterType) -> bool {
+    false
+}
```
Comment (behzadnouri) [inline turbine/src/broadcast_stage/standard_broadcast_run.rs:532]: This function will be used to incrementally roll out chained Merkle shreds starting with a small % of slots on testnet etc.

Comment (bw-solana) [reply turbine/src/broadcast_stage/standard_broadcast_run.rs:532]: Is the idea that receiving nodes won't know if chained merkle shreds are present for this slot until they deserialize shreds and look?

Here's my strawman of the rollout plan - please correct:
1. ~Plumbing for generating and including chained merkle shreds~
2. ~Start including for "non-standard" broadcast methods~
3. This PR to add single function to decide whether to chain merkle shreds for "standard" broadcast method based on slot/cluster type
4. Logic to include for some small percentage of slots on testnet
5. Receivers validating chained Merkle shreds when present
6. Expand rollout to more slots/clusters
7. Remove conditional logic
8. Receivers unconditionally verify Merkle shreds (must be present)

**Review by bw-solana — COMMENTED** [review]
LGTM but left a couple of questions

File: turbine/src/broadcast_stage/standard_broadcast_run.rs (hunk 1 above)
Comment (behzadnouri) [reply turbine/src/broadcast_stage/standard_broadcast_run.rs:532]: Yes, that is pretty much it.
We did a similar thing for the Merkle shreds rollout:
https://github.com/solana-labs/solana/pulls?q=is%3Apr+author%3Abehzadnouri+rolls+out+Merkle+is%3Aclosed+

File: turbine/src/broadcast_stage/standard_broadcast_run.rs [hunk 2]
```diff
@@ -602,18 +644,50 @@ mod test {
         let (quic_endpoint_sender, _quic_endpoint_receiver) =
             tokio::sync::mpsc::channel(/*capacity:*/ 128);
 
+        let receive_results = {
```
[the remark this answers is not in the document]
Comment (behzadnouri) [reply turbine/src/broadcast_stage/standard_broadcast_run.rs]: Reverted that change. Not needed at the moment.

**Review by bw-solana — APPROVED** [review]

**Pull request merged**